superset/assets/visualizations/deckgl/layers/geojson.jsx

```diff
@@ -14,40 +14,74 @@ const propertyMap = {
   'stroke-width': 'strokeWidth',
 };
 
-const convertGeoJsonColorProps = (p, colors) => {
-  const obj = Object.assign(...Object.keys(p).map(k => ({
-    [(propertyMap[k]) ? propertyMap[k] : k]: p[k] })));
-
+const alterProps = (props, propOverrides) => {
+  const newProps = {};
+  Object.keys(props).forEach((k) => {
+    if (k in propertyMap) {
+      newProps[propertyMap[k]] = props[k];
+    } else {
+      newProps[k] = props[k];
+    }
+  });
+  if (typeof props.fillColor === 'string') {
+    newProps.fillColor = hexToRGB(p.fillColor);
+  }
+  if (typeof props.strokeColor === 'string') {
+    newProps.strokeColor = hexToRGB(p.strokeColor);
+  }
   return {
-    ...obj,
-    fillColor: (colors.fillColor[3] !== 0) ? colors.fillColor : hexToRGB(obj.fillColor),
-    strokeColor: (colors.strokeColor[3] !== 0) ? colors.strokeColor : hexToRGB(obj.strokeColor),
+    ...newProps,
+    ...propOverrides,
   };
 };
+let features;
+const recurseGeoJson = (node, propOverrides, jsFnMutator, extraProps) => {
+  if (node && node.features) {
+    node.features.forEach((obj) => {
+      recurseGeoJson(obj, propOverrides, jsFnMutator, node.extraProps || extraProps);
+    });
+  }
+  if (node && node.properties) {
```

Comment:
I think checking for `geometry` instead of `properties` would be better here. Not all nodes need the property set, but the geometry is mandatory for defining the shape.

Reply:
good call